### ISSUE TITLE
Way to replace the Id function without changing generated code? 

### ISSUE THREAD

**Issue by manuelseeger** [odata2ts/odata2ts#144]
I have an issue with the Ids generated by odata2ts for the SAP CPI management OData API: 

https://api.sap.com/api/IntegrationContent/path/get_IntegrationDesigntimeArtifacts_Id___Id___Version___Version___

The problem: 
Resources require composite keys: 
/IntegrationDesigntimeArtifacts(Id='{Id}',Version='{Version}'

```js
// design time error since Version is not a key in metadata
const keycomplex = {
  Id: iFlowId,
  Version: "active",
};

// runtime/API error since string will get URL encoded and put in single quotes
const keysimple = `Id='${iFlowId}',Version='active'`;

const patchResult = await cpi
  .navToIntegrationDesigntimeArtifacts()
  .patch(keysimple, iFlowContent);
```
I realize the problem here is that the API is inconsistent: Some resources require keys which are not specified as keys in the metadata. I looked into replacing the Id function of the service to account for that, but have not found a way which doesn't require changing the generated code. Is there something I can do here?

I also realize this isn't exactly an issue with odata2ts - is there a development community or some place where I could raise questions like this?

**Comment by texttechne** [comment]
Hi @manuelseeger,

first off, you're completely in the right place to address issues like that. 

I try to take a look later.

**Comment by texttechne** [comment]
Cool issue, thanks for bringing it up!
So it's an instance of an imperfect OData implementation: certain entities don't advertise all of their keys in the metadata.  

I think that `odata2ts` should be able to cope with that reality of imperfect implementations.

You're going for an escape hatch: Providing the id path directly as string. 
Instead I would suggest to offer a configuration which would render such an escape hatch unnecessary.
What do you think?

## Solution Approach
Introduce config option `entitiesByName` (analogous to `propertiesByName`) to allow for further configuration of entities including their key spec.  
```ts
{
  services: {
    cpi: {
      entitiesByName: [
        { name: "IntegrationDesigntimeArtifacts", keys: ["Id", "Version"] }
      ]
    }
  }
}
```

**Comment by texttechne** [comment]
Released with `@odata2ts/odata2ts@0.20.0`.

@manuelseeger Could you try this out and confirm that it works and solves your problem?
